index.js

```diff
@@ -36,33 +38,65 @@ var gulpSass = function gulpSass(options) {
       opts.omitSourceMapUrl = true;
     }
 
-    callback = function(error, obj) {
-      if (error) {
-        return cb(new gutil.PluginError(
-          PLUGIN_NAME, error.message + ' ' + gutil.colors.cyan('line ' + error.line) + ' in ' + gutil.colors.magenta(error.file)
-        ));
-      }
+    //////////////////////////////
+    // Handles returning the file to the stream
+    //////////////////////////////
+    filePush = function filePush(sassObj) {
       // Build Source Maps!
-      if (obj.map) {
-        applySourceMap(file, JSON.parse(obj.map.toString()));
+      if (sassObj.map) {
+        applySourceMap(file, JSON.parse(sassObj.map.toString()));
       }
 
-      file.contents = obj.css;
+      file.contents = sassObj.css;
       file.path = gutil.replaceExtension(file.path, '.css');
 
       cb(null, file);
     };
 
-    sass.render(opts, callback);
+    if (sync !== true) {
+      //////////////////////////////
+      // Async Sass render
+      //////////////////////////////
+      callback = function(error, obj) {
+        if (error) {
+          return cb(new gutil.PluginError(
+            PLUGIN_NAME, error.message + ' ' + gutil.colors.cyan('line ' + error.line) + ' in ' + gutil.colors.magenta(error.file)
+          ));
+        }
+        filePush(obj);
+      };
+
+      sass.render(opts, callback);
+    }
+    else {
+      //////////////////////////////
+      // Sync Sass render
+      //////////////////////////////
+      try {
+        result = sass.renderSync(opts);
+
+        filePush(result);
+      }
+      catch(error) {
+        return cb(new gutil.PluginError(
+          PLUGIN_NAME, error.message + ' ' + gutil.colors.cyan('line ' + error.line) + ' in ' + gutil.colors.magenta(error.file)
+        ));
```

Comment:
I'd extract the error formatting as well, also not a big fan of using too many colours since you never know what colour someone is using for their terminal bg